src/store/reducers/Authentication/index.ts

```diff
@@ -1,36 +1,43 @@
 import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 
+import { getToken } from 'helpers/authentication';
 import { ErrorResponse } from 'types/error';
+import { Token } from 'types/token';
 
-import { signInAsync } from './actions';
+import { signInThunkCreator } from './actions';
+
+const token = getToken() ? getToken() : undefined;
 
 export interface AuthenticationState {
   loading: boolean;
   errors?: string[];
   success: boolean;
+  token?: Token;
 }
 
 export const initialState: AuthenticationState = {
   loading: false,
   success: false,
+  token: token,
 };
 
-export const signIn = createAsyncThunk('auth/signIn', signInAsync);
+export const signInAsyncThunk = createAsyncThunk('auth/signIn', signInThunkCreator);
```

Comment:
I think we can keep the constant name `signIn` as this is just a normal action. We rarely name the action like that.
https://redux-toolkit.js.org/api/createAsyncThunk

Reply:
@hoangmirs I have a problem when writing the unit test that `signIn` from adapter has the same name 🙈 But it's okay, let me revert it back and add the alias if there is the duplicated name